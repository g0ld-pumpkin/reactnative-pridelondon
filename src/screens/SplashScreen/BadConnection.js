// @flow
import React, { Component } from "react";
import { StyleSheet, View, ActivityIndicator } from "react-native";

import Text from "../../components/Text";
import Button from "../../components/ButtonPrimary";
import ContentPadding from "../../components/ContentPadding";

import { whiteColor, eucalyptusGreenColor } from "../../constants/colors";
import text from "../../constants/text";

type State = {
  retrying: boolean
};

type Props = {
  loading: boolean,
  getData: () => void
};

class BadConnection extends Component<Props, State> {
  constructor(props: Props) {
    super(props);
    this.state = {
      retrying: false
    };
  }

  componentWillReceiveProps({ loading }: Props) {
    if (!loading) {
      this.setState({ retrying: false });
    }
  }

  retryDataFetch = () => {
    this.setState({ retrying: true });
    this.props.getData();
  };

  render() {
    const { retrying } = this.state;
    return (
      <View style={styles.container}>
        <View>
          <Text type="h1" style={styles.title}>
            {text.badConnection.title}
          </Text>
        </View>
        <View style={styles.messageContainer}>
          <Text type="h2" style={styles.message}>
            {text.badConnection.message}
          </Text>
        </View>

        <ContentPadding style={styles.buttonContainer}>
          {!retrying && (
            <Button onPress={this.retryDataFetch}>
              {text.badConnection.button}
            </Button>
          )}
          {retrying && (
            <ActivityIndicator size="large" color={eucalyptusGreenColor} />
          )}
        </ContentPadding>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    position: "absolute",
    bottom: "10%"
  },
  title: {
    color: whiteColor,
    textAlign: "center"
  },
  message: {
    color: whiteColor,
    textAlign: "center",
    padding: 20
  },
  messageContainer: {},
  buttonContainer: {
    alignSelf: "stretch",
    minHeight: 48
  }
});

export default BadConnection;
