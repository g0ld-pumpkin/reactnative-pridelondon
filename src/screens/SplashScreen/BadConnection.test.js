// @flow
import React from "react";
import { ActivityIndicator } from "react-native";
import { shallow } from "enzyme";
import BadConnection from "./BadConnection";
import Button from "../../components/ButtonPrimary";

it("renders correctly when showing", () => {
  const output = shallow(
    <BadConnection getData={() => {}} noDataReceived={false} loading={false} />
  );

  expect(output).toMatchSnapshot();
});

it("renders correctly when with no data", () => {
  const output = shallow(
    <BadConnection getData={() => {}} noDataReceived loading={false} />
  );

  expect(output).toMatchSnapshot();
});

it("renders correctly when with no data and loading", () => {
  const output = shallow(
    <BadConnection getData={() => {}} noDataReceived loading />
  );

  expect(output).toMatchSnapshot();
});

describe("Retrying", () => {
  it("Calls getData when retry button is pressed", () => {
    const mock = jest.fn();
    const output = shallow(
      <BadConnection getData={mock} noDataReceived loading={false} />
    );

    const button = output.find(Button);
    button.props().onPress();
    expect(mock).toHaveBeenCalled();
  });

  it("Sets retrying state and renders spinner when retry button is pressed", () => {
    const mock = jest.fn();
    const output = shallow(
      <BadConnection getData={mock} noDataReceived loading={false} />
    );

    const button = output.find(Button);
    button.props().onPress();
    output.update();
    expect(output.state().retrying).toEqual(true);
    expect(output.find(ActivityIndicator).exists()).toEqual(true);
    expect(output.find(Button).exists()).toEqual(false);
  });
});
