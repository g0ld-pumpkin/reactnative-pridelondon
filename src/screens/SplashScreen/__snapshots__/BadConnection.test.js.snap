// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when showing 1`] = `
<Component
  style={
    Object {
      "bottom": "10%",
      "position": "absolute",
    }
  }
>
  <Component>
    <Text
      color="blackColor"
      markdown={false}
      markdownStyle={Object {}}
      style={
        Object {
          "color": "#ffffff",
          "textAlign": "center",
        }
      }
      type="h1"
    >
      It's not you, it's wifi!
    </Text>
  </Component>
  <Component
    style={Object {}}
  >
    <Text
      color="blackColor"
      markdown={false}
      markdownStyle={Object {}}
      style={
        Object {
          "color": "#ffffff",
          "padding": 20,
          "textAlign": "center",
        }
      }
      type="h2"
    >
      We can't find a good connection - please try again when you connection is restored
    </Text>
  </Component>
  <ContentPadding
    padding={Object {}}
    style={
      Object {
        "alignSelf": "stretch",
        "minHeight": 48,
      }
    }
  >
    <Button
      disabled={false}
      onPress={[Function]}
    >
      Try again
    </Button>
  </ContentPadding>
</Component>
`;

exports[`renders correctly when with no data 1`] = `
<Component
  style={
    Object {
      "bottom": "10%",
      "position": "absolute",
    }
  }
>
  <Component>
    <Text
      color="blackColor"
      markdown={false}
      markdownStyle={Object {}}
      style={
        Object {
          "color": "#ffffff",
          "textAlign": "center",
        }
      }
      type="h1"
    >
      It's not you, it's wifi!
    </Text>
  </Component>
  <Component
    style={Object {}}
  >
    <Text
      color="blackColor"
      markdown={false}
      markdownStyle={Object {}}
      style={
        Object {
          "color": "#ffffff",
          "padding": 20,
          "textAlign": "center",
        }
      }
      type="h2"
    >
      We can't find a good connection - please try again when you connection is restored
    </Text>
  </Component>
  <ContentPadding
    padding={Object {}}
    style={
      Object {
        "alignSelf": "stretch",
        "minHeight": 48,
      }
    }
  >
    <Button
      disabled={false}
      onPress={[Function]}
    >
      Try again
    </Button>
  </ContentPadding>
</Component>
`;

exports[`renders correctly when with no data and loading 1`] = `
<Component
  style={
    Object {
      "bottom": "10%",
      "position": "absolute",
    }
  }
>
  <Component>
    <Text
      color="blackColor"
      markdown={false}
      markdownStyle={Object {}}
      style={
        Object {
          "color": "#ffffff",
          "textAlign": "center",
        }
      }
      type="h1"
    >
      It's not you, it's wifi!
    </Text>
  </Component>
  <Component
    style={Object {}}
  >
    <Text
      color="blackColor"
      markdown={false}
      markdownStyle={Object {}}
      style={
        Object {
          "color": "#ffffff",
          "padding": 20,
          "textAlign": "center",
        }
      }
      type="h2"
    >
      We can't find a good connection - please try again when you connection is restored
    </Text>
  </Component>
  <ContentPadding
    padding={Object {}}
    style={
      Object {
        "alignSelf": "stretch",
        "minHeight": 48,
      }
    }
  >
    <Button
      disabled={false}
      onPress={[Function]}
    >
      Try again
    </Button>
  </ContentPadding>
</Component>
`;
